tests/json/generate_parsing_test.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python3
+
+import glob
+import os
+import os.path
+import sys
+
+
+def transform_string_sub(dest_file, iterator):
+    for c in iterator:
+        if c < ord(b' ') or c >= 0x7f or c == ord(b'"') or c == ord(b'\\'):
+            dest_file.write(b'\\%03o' % (c,))
+        else:
+            dest_file.write(b'%c' % (c,))
+
+
+def iter_file_as_ints(input_file):
+    while True:
+        c = input_file.read(1)
+        if not c:
+            break
+        yield ord(c)
+
+
+def transform_string(dest_file, input_file, test_name, esc_test_name):
+    dest_file.write(b'static const char ')
+    dest_file.write(esc_test_name)
+    dest_file.write(b'[] = "')
+    transform_string_sub(dest_file, iter_file_as_ints(input_file))
+    dest_file.write(b'";\n')
+
+    dest_file.write(b'static const char ')
+    dest_file.write(esc_test_name)
+    dest_file.write(b'_name[] = "')
+    transform_string_sub(dest_file, test_name)
+    dest_file.write(b'";\n')
+
+    dest_file.write(b'\n')
+
+
+def escape_test_name(test_name, index):
+    escaped_str = b''.join(
+        b'%c' % (c,) if c in range(ord('a'), ord('z') + 1) or
+        c in range(ord('A'), ord('Z') + 1) or
+        c in range(ord('0'), ord('9') + 1) and index > 0 else
+        b'_%02x' % (c,)
+        for index, c in enumerate(test_name)
+    )
+    return b'json_parsing_test_%s_%d' % (escaped_str, index)
+
+
+def main(dest_path='./generated-parsing-tests.c',
+         source_path='./JSONTestSuite',
+         cwd=os.path.dirname(os.path.realpath(__file__))):
+    test_names = []
+    cwd = os.path.abspath(cwd)
+    with open(os.path.join(cwd, dest_path), 'wb') as dest_file:
+        # License:
+        dest_file.write(b'/* This file was generated using the tests in '
+                        b'JSON Parsing Test Suite,\n')
+        dest_file.write(b' * which has the following license information:\n')
+        dest_file.write(b'\n')
+        with open(os.path.join(cwd, source_path, './LICENSE'), 'rb') as license:
```

Comment:
When starting the script without any parameters I get the following error at this line:

```
$ tests/json/generate_parsing_test.py
Traceback (most recent call last):
  File "tests/json/generate_parsing_test.py", line 124, in <module>
    result = main(*sys.argv[1:4])
  File "tests/json/generate_parsing_test.py", line 63, in main
    with open(os.path.join(cwd, source_path, './LICENSE'), 'rb') as license:
FileNotFoundError: [Errno 2] No such file or directory: '/home/mlenders/Repositories/RIOT-OS/RIOT/tests/json/./JSONTestSuite/./LICENSE'
```

I guess this needs to be downloaded somewhere, but it would be nice to have this documented somewhere (or maybe even have a script to download it)

Reply:
yep added README to provide instructions on how to update the generated stuff